docs/cvl/storage-extension-annotations.md

```diff
@@ -0,0 +1,133 @@
+(storage-extension-annotations)=
+Automatic Storage-Extension Harnesses
+=====================================
+
+Solidity ≋ EIP-7201 namespace
+-----------------------------
+
+Upgradeable contracts frequently tuck new state variables into
+namespace “bookshelves” in storage:
+
+```solidity
+/** @custom:storage-location erc7201:my.project.book1 */
+struct Book1 { /* ... */ }
+```
+
+Manually replicating the *slot math* in Certora rules or harnesses is
+tedious and error-prone.
+Lucky for you, now you can let the tool
+*auto-generate* those storage extensions.
+
+## Table of Contents
+
+- [Feature Flags](#feature-flags)
+- [How it works (high level)](#how-it-works-high-level)
+- [Quick example](#quick-example)
+- [Tips & limitations](#tips--limitations)
+- [Troubleshooting](#troubleshooting)
+
+---
+
+(feature-flags)=
+## Feature Flags
+
+To enable automatic storage extension, add one or both of the following flags to your **conf JSON**:
+
+| Flag                                           | Purpose                                                                                                                                | Default |
+| ---------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------- | ------- |
+| `"storage_extension_annotation": true         `| Detects `@custom:storage-location erc7201:…` annotations and **automatically extends the storage layout** during compilation.           | `false` |
+| `"extract_storage_extension_annotation": true` | Dumps the generated harness Solidity file(s) to `<build_dir>/…` for inspection.                            | `false` |
+
+**Example:**
+
+```json
+{
+    "storage_extension_annotation": true,
+    "extract_storage_extension_annotation": true
+}
+```
+
+(how-it-works-high-level)=
+## How it works (high level)
+
+1. **Scan ASTs** – while compiling each `.sol` file the Prover looks for
+   `StructDefinition` nodes that contain
+   `@custom:storage-location erc7201:<namespace>`.
+
+2. **Generate a minimal harness** – for every unique namespace the tool
+   emits an *extra* contract containing **only**
+
+   * Import statements for the original file,
+   * one **dummy state variable** per namespace with the prefix
+     `ext_<namespace>_` and the original struct name.
+
+     ```solidity
+     // auto-generated
+     import "./OriginalFile.sol";
+     contract _Auto_BookHarness_ {
+         Book1 ext_my_project_book1;  // slot = keccak256("erc7201:my.project.book1")-1 & ~0xff
+     }
+     ```
+
+3. **Compile the harness** with *exactly* the same `solc` flags
+   (optimizer, via-IR, remappings, --overwrite, etc.) that the main run
```

Comment:
We don't have an `--overwrite` flag. I suggest just linking to the solc flags instead in the options page

Reply:
```suggestion
   (optimizer, via-IR, remappings, --overwrite, etc.) that the main file
```